src/PowerShellEditorServices/Services/Symbols/Vistors/AstOperations.cs

```diff
@@ -92,6 +93,34 @@ static AstOperations()
                 (pwsh, _) =>
                 {
                     stopwatch.Start();
+
+                    // If the current runspace is not out of process, then we call TabExpansion2 so
+                    // that we have the ability to issue pipeline stop requests on cancellation.
+                    if (executionService is PsesInternalHost psesInternalHost
```

Comment:
Oof, I would happily remove the `IExecutionService` interface as it makes us do things like this.

Reply:
Yeah either the sync task itself needs to reference `IExecutionService` or we need to strip it out. Or maybe there should be another interface that has state or w/e.

Maybe we hold off on tackling that until we get a source generator that handles the interface definitions for us? 🤷